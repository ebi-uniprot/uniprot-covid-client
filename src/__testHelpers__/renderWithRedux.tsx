/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
