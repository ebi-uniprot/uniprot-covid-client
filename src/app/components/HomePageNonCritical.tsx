import React from 'react';
import { Tile, HeroContainer } from 'franklin-sites';

import Block from './Block';
import PlaceHolder from './PlaceHolder';

const HomePageNonCritical = () => (
  <>
    <Block columns="4">
      <Tile title="UniProtKB" />
      <Tile
        title="Proteomes"
        namespace="proteomes"
        description="Sets of proteins for reference and other species of interest."
      />
      <Tile
        title="UniRef"
        namespace="uniref"
        description="Clusters of protein sequences at 100%, 90% &amp; 50% similarity"
      />
      <Tile
        title="UniParc"
        namespace="uniparc"
        description="Non-redundant archive of protein sequences publicly available (UniProt, RefSeq, Ensembl,...)"
      />
    </Block>
    <Block columns="4">
      <Tile title="Literature Citations" small />
      <Tile title="Taxonomy" small />
      <Tile title="Subcellular locations" small />
      <Tile title="Cross-ref databases" small />
    </Block>
    <Block columns="4">
      <Tile title="Diseases" small />
      <Tile title="Keywords" small />
      <Tile title="UniRule" small />
      <Tile title="SAAS" small />
    </Block>
    <HeroContainer title="News">
      <PlaceHolder />
    </HeroContainer>
    <Block columns="4">
      <Tile
        title="BLAST"
        description="The Basic Local Alignment Search Tool finds regions of local similarity between sequences."
      />
      <Tile
        title="ALIGN"
        description="Align two or more protein sequences with the Clust Omega program."
      />
      <Tile
        title="Retrieve/ID Mapping"
        description="Retrieve and/or convert non UniProt identifiers to download lists for use on this website."
      />
      <Tile
        title="Peptide Search"
        description="Find all UniProtKB sequences that exactly match a query peptide sequence."
      />
    </Block>
    <HeroContainer title="Need Help?">
      <PlaceHolder />
    </HeroContainer>
    <Block columns="3">
      <Tile
        title="Programmatic access"
        description="Query UniProt data using APIs providing REST, SPARQL and Java services."
      />
      <Tile title="FTP Download" description="Get the UniProt data directly." />
      <Tile
        title="Technical Documentation"
        description="Access extensive technical documents to help you use UniProt to its full potential."
      />
    </Block>
  </>
);

export default HomePageNonCritical;
