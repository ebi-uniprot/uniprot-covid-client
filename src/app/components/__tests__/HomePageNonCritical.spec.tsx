import React from 'react';

import renderWithRedux from '../../../shared/__test-helpers__/RenderWithRedux';

import HomePageNonCritical from '../HomePageNonCritical';

describe('HomePage component (non-critical part)', () => {
  test('should render', () => {
    const { asFragment } = renderWithRedux(<HomePageNonCritical />);
    expect(asFragment()).toMatchSnapshot();
  });
});
