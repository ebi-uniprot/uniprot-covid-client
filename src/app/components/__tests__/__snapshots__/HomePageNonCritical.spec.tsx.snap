// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component (non-critical part) should render 1`] = `
<DocumentFragment>
  <div
    class="block block--4-col"
  >
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        UniProtKB
      </h3>
      <p
        class="tile__content"
      >
        This is a short description of what the resource is/provides.
      </p>
    </div>
    <div
      class="tile tile--proteomes"
    >
      <h3
        class="tile__header"
      >
        Proteomes
      </h3>
      <p
        class="tile__content"
      >
        Sets of proteins for reference and other species of interest.
      </p>
    </div>
    <div
      class="tile tile--uniref"
    >
      <h3
        class="tile__header"
      >
        UniRef
      </h3>
      <p
        class="tile__content"
      >
        Clusters of protein sequences at 100%, 90% & 50% similarity
      </p>
    </div>
    <div
      class="tile tile--uniparc"
    >
      <h3
        class="tile__header"
      >
        UniParc
      </h3>
      <p
        class="tile__content"
      >
        Non-redundant archive of protein sequences publicly available (UniProt, RefSeq, Ensembl,...)
      </p>
    </div>
  </div>
  <div
    class="block block--4-col"
  >
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Literature Citations
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Taxonomy
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Subcellular locations
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Cross-ref databases
      </h3>
    </div>
  </div>
  <div
    class="block block--4-col"
  >
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Diseases
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        Keywords
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        UniRule
      </h3>
    </div>
    <div
      class="tile tile--small"
    >
      <h3
        class="tile__header"
      >
        SAAS
      </h3>
    </div>
  </div>
  <div
    class="hero-container"
  >
    <section>
      <h4>
        News
      </h4>
      <div
        class="placeholder"
      >
        <test-file-stub />
      </div>
    </section>
  </div>
  <div
    class="block block--4-col"
  >
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        BLAST
      </h3>
      <p
        class="tile__content"
      >
        The Basic Local Alignment Search Tool finds regions of local similarity between sequences.
      </p>
    </div>
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        ALIGN
      </h3>
      <p
        class="tile__content"
      >
        Align two or more protein sequences with the Clust Omega program.
      </p>
    </div>
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        Retrieve/ID Mapping
      </h3>
      <p
        class="tile__content"
      >
        Retrieve and/or convert non UniProt identifiers to download lists for use on this website.
      </p>
    </div>
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        Peptide Search
      </h3>
      <p
        class="tile__content"
      >
        Find all UniProtKB sequences that exactly match a query peptide sequence.
      </p>
    </div>
  </div>
  <div
    class="hero-container"
  >
    <section>
      <h4>
        Need Help?
      </h4>
      <div
        class="placeholder"
      >
        <test-file-stub />
      </div>
    </section>
  </div>
  <div
    class="block block--3-col"
  >
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        Programmatic access
      </h3>
      <p
        class="tile__content"
      >
        Query UniProt data using APIs providing REST, SPARQL and Java services.
      </p>
    </div>
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        FTP Download
      </h3>
      <p
        class="tile__content"
      >
        Get the UniProt data directly.
      </p>
    </div>
    <div
      class="tile"
    >
      <h3
        class="tile__header"
      >
        Technical Documentation
      </h3>
      <p
        class="tile__content"
      >
        Access extensive technical documents to help you use UniProt to its full potential.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
